src/coreclr/debug/daccess/dacdbiimpl.cpp

```diff
@@ -7535,6 +7535,31 @@ HRESULT DacDbiInterfaceImpl::EnableGCNotificationEvents(BOOL fEnable)
     return hr;
 }
 
+HRESULT DacDbiInterfaceImpl::ReadData(TADDR pRemoteBuf, DWORD size, BYTE *pLocalBuf)
+{
+    DD_ENTER_MAY_THROW
+
+    HRESULT hr = S_OK;
+    EX_TRY
+    {
+        ULONG32 cbRead = 0;
+
+        HRESULT hr = m_pTarget->ReadVirtual(pRemoteBuf, pLocalBuf, size, &cbRead);
+
+        if (FAILED(hr))
+        {
+            ThrowHR(CORDBG_E_READVIRTUAL_FAILURE);
+        }
+
+        if (cbRead != size)
+        {
+            ThrowWin32(ERROR_PARTIAL_COPY);
+        }
+    }
+    EX_CATCH_HRESULT(hr);
```

Comment:
If you change the above throws to returns, then you don't need the EX_TRY/EX_CATCH_HRESULT.  ReadVirtual should never throw an exception.